dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/TextCompletionStreamingResult.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.AzureSdk;
+
+internal sealed class TextCompletionStreamingResult : ITextCompletionStreamingResult
+{
+    private readonly StreamingChoice _choice;
+
+    public TextCompletionStreamingResult(StreamingChoice choice)
+    {
+        this._choice = choice;
+    }
+
+    public async Task<string> CompleteAsync(CancellationToken cancellationToken = default)
+    {
+        var fullMessage = new StringBuilder();
+        await foreach (var message in this._choice.GetTextStreaming(cancellationToken).ConfigureAwait(false))
```

Comment:
Will calling GetTextStreaming return the same results twice? If someone calls TextCompletionStreamingResult.CompleteAsync() twice, do they get the streaming result once, and the completed result the second time?

Reply:
That's right, made a double check test here. 

The result is kept in memory after getting from the model.

So, if someone calls for the same `Result` instance a second time (either Streaming or not) it will get the same response.

ℹ️ Any further call on finished streaming result will be almost instant.

Reply:
Thanks!